custom-plugins/cms131v14_diabetes_eye_exam/protocols/cms131v14_protocol.py

```diff
@@ -0,0 +1,690 @@
+        """
+        Check if patient is in hospice or palliative care during the measurement period.
+
+        Checks for:
+        - Referral reports to hospice/palliative care services
+        - Active condition codings against hospice value sets
+        """
+        try:
+            # Check for hospice/palliative care referral reports
+            hospice_reports = ReferralReport.objects.filter(
+                patient=patient,
+                junked=False,
+                original_date__gte=self.timeframe.start.date(),
+                original_date__lte=self.timeframe.end.date(),
+            ).all()
+
+            # Check if any reports are for hospice specialties
+            hospice_specialties = {"Hospice", "Palliative Care", "Palliative Medicine"}
+            for report in hospice_reports:
+                if report.specialty and any(
+                    hospice_specialty.lower() in report.specialty.lower()
+                    for hospice_specialty in hospice_specialties
+                ):
+                    log.info(
+                        f"CMS131v14: Found hospice referral report for patient {patient.id}"
+                    )
+                    return True
+
+            # Additionally check condition codings against hospice value sets
+            return self._patient_has_any_code_in_value_sets(
+                patient,
+                [PalliativeOrHospiceCare, HospiceCareAmbulatory],
+            )
+        except Exception as e:
+            log.error(f"CMS131v14: Error checking hospice status: {str(e)}")
+            return False
+
+    def _is_age_66_plus_with_frailty(self, patient: Patient) -> bool:
+        """
+        Check if patient is age 66+ with frailty indicators.
+        Per CMS131v14: This exclusion only applies to patients age 66 and older.
+        """
+        # Check age requirement
+        age = self._calculate_age(patient)
+        if age is None or age < 66:
+            return False
+
+        try:
+            # Check for frailty indicators
+            has_frailty = self._patient_has_any_code_in_value_sets(
+                patient,
+                [FrailtySymptom, FrailtyDevice, FrailtyEncounter],
+            )
+
+            if has_frailty:
+                log.info(f"CMS131v14: Patient {patient.id} age 66+ has frailty indicators")
+
+            return has_frailty
+        except Exception as e:
+            log.error(f"CMS131v14: Error checking frailty: {str(e)}")
+            return False
+
+    def _has_advanced_illness_or_dementia_meds(self, patient: Patient) -> bool:
+        """
+        Check if patient has advanced illness (dementia, cancer, chemotherapy) or HCPCS S0311.
+        """
+        try:
+            # Check for advanced illness conditions and medications
+            has_advanced_illness = self._patient_has_any_code_in_value_sets(
+                patient,
+                [DementiaAndMentalDegenerations, Cancer, ChemotherapyForAdvancedCancer],
+            )
```

Comment:
This says it's checking for the patient to be on some medications, but is only checking conditions.